components/menu/index.tsx

```diff
@@ -42,9 +42,8 @@ type CompoundedComponent = React.ForwardRefExoticComponent<GenericComponentProps
   ItemGroup: typeof ItemGroup;
 };
 
-interface GenericComponent extends Omit<CompoundedComponent, ''> {
-  <T extends MenuItemType>(props: GenericComponentProps<T>): ReturnType<CompoundedComponent>;
-}
+type GenericComponent = Omit<CompoundedComponent, ''> &
+  (<T extends MenuItemType>(props: GenericComponentProps<T>) => ReturnType<CompoundedComponent>);
```

Comment:
Could you add type test in `components/menu/__tests__/type.test.tsx`?

Reply:
I am not sure how i would do that since typescript only complains when the variable is re-exported

Reply:
Why make this Change and what are the benefits?

Reply:
When it's an interface and its not exported, typescript complains about the usage of it. Similar problem and a merged pr to fix it is referenced:
https://github.com/ant-design/ant-design/issues/42907
https://github.com/ant-design/ant-design/issues/42907

Reply:
> When it's an interface and its not exported, typescript complains about the usage of it

How can I reproduce it?

Reply:
https://codesandbox.io/p/sandbox/979dqs

@afc163 See the CustomMenu component. tsconfig.json file is from apache superset